pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -305,6 +311,18 @@ func (state PipelineRunState) GetTaskRunsStatus(pr *v1beta1.PipelineRun) map[str
 				})
 			}
 		}
+		// Maintain a TaskRun Object in pr.Status for a finally task which could not resolve task results
```

Comment:
saw that this was already agreed upon in the TEP, but I think it might be worthwhile to explore adding these final tasks to skipped tasks (or another separate section for final tasks with invalid results) -- instead of creating a TaskRun object for a final task that wasn't executed 

related discussion: https://github.com/tektoncd/pipeline/issues/3139

Reply:
yes thats an alternative, another alternative could be: Tekton controller replacing such unresolved param with type default which is empty string for string type params. Let that final task execute expecting that the task author has implemented sanitization on such input params. @bobcatfish thoughts?